lib/ansible/modules/cloud/podman/podman_container.py

```diff
@@ -0,0 +1,1350 @@
+            cmd += ['--subuidname', params['subuidname']]
+
+        if params['sysctl']:
+            cmd += ['--sysctl', params['sysctl']]
+
+        if params['systemd'] is not None:
+            cmd += ['--systemd', params['systemd']]
+
+        if params['tmpfs']:
+            cmd += ['--tmpfs', params['tmpfs']]
+
+        if params['tty'] is not None:
+            cmd += ['--tty', params['tty']]
+
+        if params['uidmap']:
+            cmd += ['--uidmap', params['uidmap']]
+
+        if params['ulimit']:
+            cmd += ['--ulimit', params['ulimit']]
+
+        if params['user']:
+            cmd += ['--user', params['user']]
+
+        if params['userns']:
+            cmd += ['--userns', params['userns']]
+
+        if params['uts']:
+            cmd += ['--uts', params['uts']]
+
+        for vol in params['volume']:
+            cmd += ['--volume', vol]
+
+        for vol in params['volumes_from']:
+            cmd += ['--volumes-from', vol]
+
+        if params['workdir']:
+            cmd += ['--workdir', params['workdir']]
+
+        # Add your own args for podman command
+        if params['cmd_args']:
+            cmd += params['cmd_args']
+
+        cmd.append(params['image'])
+
+        if params['command']:
+            cmd += params['command'].split()
+
+        return [to_bytes(i, errors='surrogate_or_strict') for i in cmd]
+
+
+def ensure_image_exists(module, image):
+    """If image is passed, ensure it exists, if not - pull it or fail
+
+    Arguments:
+        module {obj} -- ansible module object
+        image {str} -- name of image
+
+    Returns:
+        list -- list of image actions - if it pulled or nothing was done
+    """
+    image_actions = []
+    if not image:
+        return image_actions
+    rc, out, err = run_podman_command(module, args=['image', 'exists', image], ignore_errors=True)
+    if rc == 0:
+        return image_actions
+    rc, out, err = run_podman_command(module, args=['image', 'pull', image], ignore_errors=True)
+    if rc != 0:
+        module.fail_json(msg="Can't pull image %s" % image, stdout=out, stderr=err)
+    image_actions.append("pulled image %s" % image)
+    return image_actions
+
+
+class PodmanContainer:
+    """Perform container tasks
+
+    Manages podman container, inspects it and checks its current state
+    """
+    def __init__(self, module, name):
+        """Initialize PodmanContainer class
+
+        Arguments:
+            module {obj} -- ansible module object
+            name {str} -- name of container
+        """
+
+        super(PodmanContainer, self).__init__()
+        self.module = module
+        self.name = name
+        self.info = self.get_info()
+
+    @property
+    def exists(self):
+        """Check if container exists """
+        return bool(self.info != {})
+
+    @property
+    def different(self):
+        """Check if container is different"""
+        # TODO(sshnaidm): implement difference calculation between input vars
```

Comment:
Ok, so essentially idempotency is missing completely?

Reply:
Currently yes, I started to work on comparing containers to make 'started'/'present' options idempotent.